crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -0,0 +1,245 @@
+    tx!(
+        solver.account(),
+        onchain
+            .contracts()
+            .uniswap_v2_factory
+            .create_pair(token_gno.address(), token_dai.address())
+    );
+    tx!(
+        solver.account(),
+        token_gno.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        token_dai.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        onchain.contracts().uniswap_v2_router.add_liquidity(
+            token_gno.address(),
+            token_dai.address(),
+            to_wei(1000),
+            to_wei(1000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Approve GPv2 for trading
+    tx!(
+        trader.account(),
+        token_gno.approve(onchain.contracts().allowance, to_wei(100))
+    );
+
+    // Place Orders
+    let services = Services::new(onchain.contracts()).await;
+    let solver_endpoint =
+        colocation::start_baseline_solver(onchain.contracts().weth.address()).await;
+    colocation::start_driver(
+        onchain.contracts(),
+        vec![SolverEngine {
+            name: "test_solver".into(),
+            account: solver,
+            endpoint: solver_endpoint,
+        }],
+    );
+    services.start_autopilot(vec![
+        "--drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        "--fee-policy-skip-market-orders=false".to_string(),
+        fee_policy,
+    ]);
+    services
+        .start_api(vec![
+            "--price-estimation-drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        ])
+        .await;
+
+    let order = OrderCreation {
+        sell_token: token_gno.address(),
+        sell_amount: to_wei(10),
+        buy_token: token_dai.address(),
+        buy_amount: to_wei(5),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: order_kind,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+    let uid = services.create_order(&order).await.unwrap();
+
+    // Drive solution
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 1 })
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 0 })
+        .await
+        .unwrap();
+
+    onchain.mint_blocks_past_reorg_threshold().await;
+    let metadata_updated = || async {
+        onchain.mint_block().await;
+        let order = services.get_order(&uid).await.unwrap();
+        are_close_enough(order.metadata.executed_surplus_fee, expected_surplus_fee)
```

Comment:
Would be nice if we'd also test that the settlement contract actually kept the fee tokens in it.

Reply:
Very nice suggestion! Added.